### PR TITLE
[TRAFODION-1605] sqstart hangs if ulimits not correct

### PR DESCRIPTION
**Pull request by traflm**
add ulimit checking in sqstart
if not proper setup as Trafodion installer does, print WARNING/ERROR

This should only happen in dev environment as described in the JIRA. Installer already handle this, so a WARNING/ERROR print should be enough.

### CODE REVIEW

**Comment by selvaganesang** [comment]
Change looks good

File: core/sqf/sql/scripts/sqstart [hunk 1]
```diff
@@ -245,6 +245,39 @@ function checkKerberos {
...
+    echoLog "ERROR: detect issue during startup, please make sure your ulimit setup correctly"
+    echoLog "ERROR: max locked memory is smaller than 64M"
+    echoLog ""
+    return 
+ fi
+  
+ #check ulimit -l
+ if [[ $maxLockMem -lt 327680 ]] ; then
+    echoLog ""
+    echoLog "WARNING: detect issue during startup, please make sure your ulimit setup correctly"
```
Comment (DaveBirdsall) [inline core/sqf/sql/scripts/sqstart]: This message mentions ulimit, but it is really about max locked memory.

Also, consider: "WARNING: issue detected during startup, please make sure your max locked memory is set up correctly."

**Comment by DaveBirdsall** [comment]
One of the error messages is incorrect. I also suggested some rephrasing of other error messages to make the English a bit smoother. Other than that, the changes look good to me.

File: core/sqf/sql/scripts/sqstart [hunk 2]
```diff
@@ -245,6 +245,39 @@ function checkKerberos {
    return 0
 }
 
+function checkUlimit {
+  
+ #check max_pid 
+ userProc=`cat /proc/sys/kernel/pid_max`
+ if [[ $userProc -gt 65535 ]] ; then
+    echoLog ""
+    echoLog "ERROR: detect issue during startup, please make sure your kernel parameter (sysctl) setup correctly"
```
Comment (DaveBirdsall) [inline core/sqf/sql/scripts/sqstart]: Suggested wordsmith: "Error: issue detected during startup. Please make sure your kernel parameter (sysctl) is set up correctly."

File: core/sqf/sql/scripts/sqstart [hunk 3]
```diff
@@ -245,6 +245,39 @@ function checkKerberos {
...
+    echoLog "ERROR: max_pid should not be greater than 65535"
+    echoLog ""
+    return
+ fi
+
+ #check ulimit -l
+ maxLockMem=`ulimit -l`
+ if [[ $maxLockMem -lt 65536 ]] ; then
+    echoLog ""
+    echoLog "ERROR: detect issue during startup, please make sure your ulimit setup correctly"
```
Comment (DaveBirdsall) [inline core/sqf/sql/scripts/sqstart]: Suggested wordsmith: "ERROR: issue detected during startup. Please make sure your ulimit is set up correctly."

Comment (traflm) [inline core/sqf/sql/scripts/sqstart]: Many thanks for the correct English words, I will update it.

File: core/sqf/sql/scripts/sqstart (hunk 1 above)
Comment (traflm) [inline core/sqf/sql/scripts/sqstart]: hi, Dave, I think this is ulimit setting. 'max lock memory' can be set via 'ulimit -l'.  It is same number check as above. If it is smaller than 64M, it is an Error, if it is bigger than 64M, but smaller than 320M, it is a Warning, as suggested by @selvaganesang . 
But I will update the words with what you mentioned below.

Comment (DaveBirdsall) [inline core/sqf/sql/scripts/sqstart]: Ah, I see. Thanks for the explanation. Perhaps "WARNING: issue detected during startup. Please make sure ulimit -l is set up correctly."

**Comment by DaveBirdsall** [comment]
Looks good. Thanks @traflm!

**Pull request merged**